graylog2-server/src/main/java/org/graylog/plugins/pipelineprocessor/rest/RuleResource.java

```diff
@@ -173,14 +188,50 @@ public RuleSource parse(@ApiParam(name = "rule", required = true) @NotNull RuleS
                 .build();
     }
 
+    @ApiOperation(value = "Simulate a single processing rule")
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/simulate")
+    @NoAuditEvent("only used to test a rule, no changes made in the system")
+    public Message simulate(
+            @ApiParam(name = "request", required = true) @NotNull SimulateRuleRequest request
+    ) throws NotFoundException {
+        final Rule rule = parseRuleOrThrow(request.ruleSource().id(), request.ruleSource().source(), true);
+        Stage stage = Stage.builder()
+                .stage(0)
+                .ruleReferences(Collections.emptyList())
+                .match(Stage.Match.PASS)
+                .build();
+        stage.setRules(List.of(rule));
+        Pipeline pipeline = Pipeline.builder()
+                .stages(ImmutableSortedSet.of(stage))
+                .name("dummyPipeline")
+                .id(UUID.randomUUID().toString())
+                .build();
+        stage.setPipeline(pipeline);
+
+        PipelineInterpreter pipelineInterpreter = new PipelineInterpreter(
+                new NoopMessageQueueAcknowledger(), MetricRegistryFactory.create(), configurationStateUpdater);
+        final PipelineInterpreterTracer pipelineInterpreterTracer = new PipelineInterpreterTracer();
+
+        Message message = new Message(request.message(), "127.0.0.1", DateTime.now(DateTimeZone.UTC));
+//        final Stream stream = streamService.load(Stream.DEFAULT_STREAM_ID);
+//        message.addStream(stream);
```

Comment:
Is there a reason to comment out this code instead of removing it?

Reply:
Forgot to remove it after testing. Gone now.